src/zhinst/toolkit/nodetree/node.py

```diff
@@ -1108,3 +1120,47 @@ def __eq__(self, other):
 
     def __hash__(self):
         return Node.__hash__(self)
+
+
+class WildcardResult(Mapping):
+    """Mapping of results when a :class:`Node` is called with wildcards.
+
```

Comment:
Maybe add a comment why it this is necessary.

e.g. The wildcard results are a custom mapping class to allow accessing element both by string and Node object